dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCApplicationMBeanStatusTest.java

```diff
@@ -0,0 +1,104 @@
+ * TestBadServletContextListener.war web application
+ * which is expected to fail to start due to runtime exception.
+ *
+ * PASS: if status is INSTALLED
+ *
+ */
+@RunWith(FATRunner.class)
+public class WCApplicationMBeanStatusTest extends LoggingTest {
+
+    private final static String CLASSNAME = WCApplicationMBeanStatusTest.class.getName();
+
+    private final static Logger LOG = Logger.getLogger(CLASSNAME);
+
+    @ClassRule
+    public static SharedServer SHARED_SERVER = new SharedServer("servlet40_stopAppStartListenerExceptionServer");
+
+    @Override
+    protected SharedServer getSharedServer() {
+        return SHARED_SERVER;
+    }
+
+    /**
+     * @throws Exception
+     *
+     *                       The test case expects start of the application fails due to runtime
+     *                       exception on server start, add the expected ffdc.
+     */
+    @BeforeClass
+    @ExpectedFFDC({ "java.lang.Exception", "java.lang.RuntimeException" })
+    public static void setUp() throws Exception {
+        LOG.info("Setup : add TestBadServletContextListener.war to the server if not already present.");
+
+        WCApplicationHelper.addWarToServerDropins(SHARED_SERVER.getLibertyServer(), "TestBadServletContextListener.war", false,
+                                                  "testbadscl.war.listener");
+        LOG.info("Setup : TestBadServletContextListener.war added to the server");
+
+        WCApplicationHelper.addWarToServerDropins(SHARED_SERVER.getLibertyServer(), "TestServlet40.war", true, "testservlet40.war.servlets");
+
+        LOG.info("Setup : TestServlet40.war added to the server");
+
+        SHARED_SERVER.startIfNotStarted();
+
+        // The ApplicationMBean is added in TestServlet40
+        WCApplicationHelper.waitForAppStart("TestServlet40", WCPushBuilderTest.class.getName(), SHARED_SERVER.getLibertyServer());
+
+        LOG.info("Setup : TestServlet40.war started on the server");
+
+        // The failing servletcontext is added in TestBadServletContextListener
+        WCApplicationHelper.waitForAppFailStart("TestBadServletContextListener", CLASSNAME,
+                                                SHARED_SERVER.getLibertyServer());
+
+        LOG.info("Setup : TestBadServletContextListener failed to start on the server as expected");
+
+        LOG.info("Setup : complete, ready for Tests");
+    }
+
+    /**
+     * @throws Exception
+     *
+     *                       The test case expects start of the application fails due to runtime
+     *                       exception in listener, add the expected errors
+     *
+     */
+    @AfterClass
+    public static void testCleanup() throws Exception {
+        // test cleanup
+        if (SHARED_SERVER.getLibertyServer() != null && SHARED_SERVER.getLibertyServer().isStarted()) {
+            SHARED_SERVER.getLibertyServer().stopServer("SRVE0283E:.*", "SRVE0265E:.*", "CWWKE0921W:.*", "CWWKE0912W:.*", "SRVE9014E:.*");
+        }
```

Comment:
What are these Error/Warning messages for? Are these part of this specific test case?